src/libraries/Common/src/Interop/Android/System.Security.Cryptography.Native.Android/Interop.X509.cs

```diff
@@ -32,6 +32,18 @@ internal static byte[] X509Encode(SafeX509Handle x)
 
             return encoded;
         }
+        [LibraryImport(Libraries.AndroidCryptoNative, EntryPoint = "AndroidCryptoNative_X509IsKeyStorePrivateKeyEntry")]
+        [return: MarshalAs(UnmanagedType.U1)]
+        internal static partial bool IsKeyStorePrivateKeyEntry(IntPtr handle);
+        [LibraryImport(Libraries.AndroidCryptoNative, EntryPoint = "AndroidCryptoNative_X509GetCertificateForPrivateKeyEntry")]
+        private static partial IntPtr GetPrivateKeyEntryCertificate(IntPtr privatKeyEntryHandle);
+        internal static SafeX509Handle GetPrivateKeyEntryCertificate(SafeHandle privatKeyEntryHandle)
+        {
+            var certificateHandle = new SafeX509Handle();
+            var certificatePtr = GetPrivateKeyEntryCertificate(privatKeyEntryHandle.DangerousGetHandle());
```

Comment:
This should use an explicit type, not `var` here.

Reply:
What is ensuring the liveliness of the handle here? Do we need to do a `DangerousAddRef` and `DangerousRelease` around getting the handle? This is of particular concern if multiple threads are involved.

To ask another way, can `privatKeyEntryHandle` be disposed between the calls of `DangerousGetHandle` and `GetPrivateKeyEntryCertificate`?

Reply:
I added `DangerousAddRef`+`DangerousRelease` to make sure the handle isn't disposed by another thread.